### ISSUE TITLE
OPENDMARC_LIB_VERSION no longer contains version information

### ISSUE THREAD

**Issue by DrRemorse** [trusteddomainproject/OpenDMARC#235]
This used to contain version information:

$ grep OPENDMARC_LIB_VERSION /usr/include/opendmarc/dmarc.h
#define OPENDMARC_LIB_VERSION   0x01030200

Now it contains 0x00000000.  Can you check to see if perhaps a version number generator is not working?

I am currently using the following logic and hope version numbers are supported again in the future:

#if (OPENDMARC_LIB_VERSION >= 0x01040000L) || (OPENDMARC_LIB_VERSION == 0x00000000L)
  s = opendmarc_policy_store_dkim(dmarc, domain, selector, dkim_to_dmarc(r), comment);
#else
  s = opendmarc_policy_store_dkim(dmarc, domain, dkim_to_dmarc(r), comment);
#endif

Thank you!

**Comment by kjc75** [comment]
Appears this used to be set as part of the `dist-hook` `Makefile` target, but that's not getting triggered via the GitHub release process, and winds up remaining the default 0.  I don't see it set in any of the 1.4 release packages on here.

**Comment by thegushi** [comment]
This has been fixed in PR #301 on the `fix/lib-version-235` branch.

The root cause: `OPENDMARC_LIB_VERSION` was hardcoded as `0x00000000` in `dmarc.h`, with the substitution deferred to a `dist-hook` that ran during `make dist`. GitHub releases pull a raw git tarball and bypass `make dist` entirely, so the hook never fired.

The fix converts `libopendmarc/dmarc.h` to an autoconf template (`dmarc.h.in`) so that `./configure` stamps the version correctly at build time. This is the right fix for the primary consumers of OpenDMARC: OS packagers (Debian, Fedora, etc.) already run `./configure` as part of their packaging workflow, so they will get a correctly-stamped `dmarc.h` automatically once this lands. For version 1.4.2 it produces `0x01040200`.

**Comment by thegushi** [comment]
Fixed in two parts: #301 fixed the release tarball (generated header was absent); #363 removes the tracked dmarc.h from the repository (it was committed with 0x00000000, masking the configure substitution). See also issue #167.
